### PR TITLE
Move wire check to the tx validation logic

### PR DESCRIPTION
**Pull request by georgeee**
Problem: this check is the only one we perform in this layer, seemingly for no good reason.

Solution: move the check to where all of the other zkapp command checks reside.

P.S. this refactoring makes analysis of the code easier, because all tx-related checks are now located in a single place.

Explain how you tested your changes:
* Tested on mainnet via custom build

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? None

### CODE REVIEW

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

File: src/lib/mina_base/zkapp_command.ml [hunk 1]
```diff
@@ -1576,21 +1570,20 @@ let arg_query_string x =
   Fields_derivers_zkapps.Test.Loop.json_to_string_gql @@ to_json x
 
 let dummy =
-  lazy
```
Comment (mrmr1993) [inline src/lib/mina_base/zkapp_command.ml:1579]: This is an unrelated change. Please revert it.

**Review by mrmr1993 — CHANGES_REQUESTED** [review]

File: src/lib/mina_base/zkapp_command.ml (hunk 1 above)
Comment (georgeee) [reply src/lib/mina_base/zkapp_command.ml:1579]: @mrmr1993 would you advise taking this change to a separate PR or completely removing it w/o any follow-up?

**Comment by georgeee** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Review by gwapIgot — APPROVED** [review]
Excellent
